### PR TITLE
Add CRDB-based scd.Store implementation

### PR DESCRIPTION
**Pull request by vosst**

### CODE REVIEW

File: pkg/dss/scd/models/operations.go [hunk 1]
```diff
@@ -35,3 +36,39 @@ type Operation struct {
...
+			o.StartTime = old.StartTime
+		}
+	} else {
+		// If setting the StartTime explicitly ensure it is not too far in the past.
+		if now.Sub(*o.StartTime) > maxClockSkew {
+			return dsserr.BadRequest("Operation time_start must not be in the past")
+		}
+	}
+
+	// If EndTime was omitted default to the existing ISA's EndTime.
```
Comment (BenjaminPelletier) [inline pkg/dss/scd/models/operations.go:58]: Let's update "ISA"

File: pkg/dss/scd/models/operations.go [hunk 2]
```diff
@@ -35,3 +36,39 @@ type Operation struct {
...
+	if o.EndTime == nil && old != nil {
+		o.EndTime = old.EndTime
+	}
+
+	// EndTime cannot be omitted for new ISAs.
```
Comment (BenjaminPelletier) [inline pkg/dss/scd/models/operations.go:63]: Update ISA

File: pkg/dss/scd/models/operations.go [hunk 3]
```diff
@@ -35,3 +36,39 @@ type Operation struct {
 	Cells          s2.CellUnion
 	SubscriptionID ID
 }
+
+// AdjustTimeRange adjusts the time range to the max allowed ranges on an
+// Operation.
+func (o *Operation) AdjustTimeRange(now time.Time, old *Operation) error {
+	if o.StartTime == nil {
+		// If StartTime was omitted, default to Now() for new ISAs or re-
```
Comment (BenjaminPelletier) [inline pkg/dss/scd/models/operations.go:44]: I think we actually don't want to default to now for Operations; we should just reject if the start time isn't specified.

File: pkg/dss/scd/models/operations.go [hunk 4]
```diff
@@ -35,3 +36,39 @@ type Operation struct {
...
+		// use the existing time of existing ISAs.
+		if old == nil {
+			o.StartTime = &now
+		} else {
+			o.StartTime = old.StartTime
+		}
+	} else {
+		// If setting the StartTime explicitly ensure it is not too far in the past.
```
Comment (BenjaminPelletier) [inline pkg/dss/scd/models/operations.go:52]: The start time for Operations can be in the past

File: pkg/dss/scd/models/operations.go (hunk 3 above)
Comment (vosst) [reply pkg/dss/scd/models/operations.go]: Done.

File: pkg/dss/scd/models/operations.go (hunk 4 above)
Comment (vosst) [reply pkg/dss/scd/models/operations.go]: Done.

File: pkg/dss/scd/models/operations.go (hunk 1 above)
Comment (vosst) [reply pkg/dss/scd/models/operations.go]: Done.

File: pkg/dss/scd/models/operations.go (hunk 2 above)
Comment (vosst) [reply pkg/dss/scd/models/operations.go]: Done.

**Pull request merged**